lambdas/data-migration1/src/collections.ts

```diff
@@ -0,0 +1,131 @@
+import Knex from 'knex';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import { envUtils } from '@cumulus/common';
+import Logger from '@cumulus/logger';
+
+import { RecordAlreadyMigrated } from './errors';
+import { MigrationSummary } from './types';
+
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+
+const logger = new Logger({ sender: '@cumulus/data-migration/collections' });
+
+export interface RDSCollectionRecord {
+  name: string
+  version: string
+  process: string
+  granuleIdValidationRegex: string
+  granuleIdExtractionRegex: string
+  files: string
+  // default will be set by schema validation
+  duplicateHandling: string
+  // default will be set by schema validation
+  reportToEms: boolean
+  sampleFileName?: string
+  url_path?: string
+  ignoreFilesConfigForDiscovery?: boolean
+  meta?: object
+  tags?: string
+  created_at: Date
+  updated_at: Date
+}
+
+/**
+ * Migrate collection record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Source record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordAlreadyMigrated}
+ *   if record was already migrated
+ */
+export const migrateCollectionRecord = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<number> => {
```

Comment:
It is used by the tests. Is there a better way?

https://github.com/nasa/cumulus/blob/master/lambdas/data-migration1/tests/test-collections.js#L138

Reply:
In the tests, rather than:

```js
const [createdRecord] = await t.context.knex.queryBuilder()
    .select()
    .table('collections')
    .where('cumulusId', cumulusId);
```

you could do something like:

```js
const [createdRecord] = await t.context.knex.queryBuilder()
    .select()
    .table('collections')
    .where({ name: fakeCollection.name, version: fakeCollection.version });
```

Reply:
🤦 Indeed

Reply:
9c20c460168bf02cf59826e5f2bf64106bf8f23c